src/Discord.Net/Utilities/EmoteUtilities.cs

```diff
@@ -7,21 +7,28 @@ public static class EmoteUtilities
         public static string FormatGuildEmote(ulong id, string name)
             => $"<:{name}:{id}>";
 
-        public static (ulong, string) ParseGuildEmote(string formatted)
+        // TODO: perf: bench whether this should be passed by ref (in)
+        public static bool TryParseGuildEmote(ReadOnlySpan<char> formatted, out (ulong, string) result)
```

Comment:
I'd suggest returning a ReadOnlySpan<char> instead of a string to avoid allocating another string here. If the user needs a string, they can easily allocate one themselves.